algorithms/src/snark/marlin/ahp/indexer/indexer.rs

```diff
@@ -194,27 +191,28 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
         let non_zero_c_domain =
             EvaluationDomain::new(num_non_zero_c).ok_or(SynthesisError::PolynomialDegreeTooLarge)?;
 
-        let (constraint_domain_elements, constraint_domain_eq_poly_vals) =
-            precomputation_for_matrix_evals(&constraint_domain);
+        let constraint_domain_elements = constraint_domain.elements().collect::<Vec<_>>();
+        let variable_domain_elements = variable_domain.elements().collect::<Vec<_>>();
 
         let [a_evals, b_evals, c_evals]: [_; 3] =
             cfg_into_iter!([(&a, &non_zero_a_domain), (&b, &non_zero_b_domain), (&c, &non_zero_c_domain),])
                 .map(|(matrix, non_zero_domain)| {
                     matrix_evals(
                         matrix,
                         non_zero_domain,
```

Comment:
shouldn't we also be passing in `constraint_domain` here, as that indexes the rows?

Reply:
Yeah this assymetry seemed weird to me as well. But the only reason we use the `variable_domain`, is because of how we're storing variable indexes in a locally correct format: `variable_domain.reindex_by_subdomain(input_domain, *input_var_index)`.